paddle/operators/bipartite_match_op.cc

```diff
@@ -0,0 +1,178 @@
+      int max_row_idx = -1;
+      T max_dis = -1;
+      for (int64_t j = 0; j < col; ++j) {
+        if (match_indices[j] != -1) {
+          continue;
+        }
+        for (int k = 0; k < row_pool.size(); ++k) {
+          int m = row_pool[k];
+          // distance is 0 between m-th row and j-th column
+          if (dis_data[m * col + j] < 1e-6) {
+            continue;
+          }
+          if (dis_data[m * col + j] > max_dis) {
+            max_idx = j;
+            max_row_idx = m;
+            max_dis = dis_data[m * col + j];
+          }
+        }
+      }
+      if (max_idx == -1) {
+        // Cannot find good match.
+        break;
+      } else {
+        PADDLE_ENFORCE_EQ(match_indices[max_idx], -1);
+        match_indices[max_idx] = max_row_idx;
+        match_dis[max_idx] = max_dis;
+        // Erase the row index.
+        row_pool.erase(
+            std::find(row_pool.begin(), row_pool.end(), max_row_idx));
+      }
+    }
+  }
+
+  void Compute(const framework::ExecutionContext& context) const override {
+    auto* dis_mat = context.Input<LoDTensor>("DisMat");
+    auto* match_indices = context.Output<Tensor>("ColToRowMatchIndices");
+    auto* match_dis = context.Output<Tensor>("ColToRowMatchDis");
+
+    auto& dev_ctx = context.device_context<platform::CPUDeviceContext>();
+
+    auto col = dis_mat->dims()[1];
+
+    int64_t n = dis_mat->lod().size() == 0
+                    ? 1
+                    : static_cast<int64_t>(dis_mat->lod().back().size() - 1);
+    match_indices->mutable_data<int>({n, col}, context.GetPlace());
+    match_dis->mutable_data<T>({n, col}, context.GetPlace());
+
+    math::SetConstant<platform::CPUDeviceContext, int> iset;
+    iset(dev_ctx, match_indices, static_cast<int>(-1));
+    math::SetConstant<platform::CPUDeviceContext, T> tset;
+    tset(dev_ctx, match_dis, static_cast<T>(0));
+
+    int* indices = match_indices->data<int>();
+    T* dis = match_dis->data<T>();
+    if (n == 1) {
+      BipartiteMatch(*dis_mat, indices, dis);
+    } else {
+      auto lod = dis_mat->lod().back();
+      for (size_t i = 0; i < lod.size() - 1; ++i) {
+        Tensor one_ins = dis_mat->Slice(lod[i], lod[i + 1]);
+        BipartiteMatch(one_ins, indices + i * col, dis + i * col);
+      }
+    }
+  }
+};
+
+class BipartiteMatchOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  BipartiteMatchOpMaker(OpProto* proto, OpAttrChecker* op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput(
+        "DisMat",
+        "(LoDTensor or Tensor) this input is a 2-D LoDTensor with shape "
+        "[K, M]. It is pair-wise distance matrix between the entities "
+        "represented by each row and each column. For example, assumed one "
+        "entity is A with shape [K], another entity is B with shape [M]. The "
+        "DisMat[i][j] is the distance between A[i] and B[j]. The bigger "
+        "the distance is, the more similar the pairs are. Please note, "
```

Comment:
We need a better description here. `The bigger the distance is, the more similar the pairs are.` seems not reasonable.